### ISSUE TITLE
`show-point-mode` is incompatible with this package

### ISSUE THREAD

**Issue by wasamasa** [Bruce-Connor/smart-mode-line#112]
I've found the need for displaying the value of `(point)` in my mode line after learning a good bit about text properties and manipulation in Emacs, so to avoid reinventing the wheel I installed [show-point-mode](https://github.com/emacsmirror/show-point-mode).

Unfortunately `show-point-mode` directly manipulates what appears to be Emacs' default display of the current line and column when using `line-number-mode` and `column-number-mode` which looks pretty wrong in combination with `smart-mode-line` since I get the display of current line and column twice plus the value of point.

Can this be fixed in a meaningful manner? Or do I need to NIH my own rendition of `show-point-mode`?

**Comment by Bruce-Connor** [comment]
First of all, here's a quick patch:

    (add-to-list 'mode-line-position '(show-point-mode (5 (:eval (format "(%d)" (point))))) 'append)

This is mostly a bug in how `show-point-mode` does its thing. It tries to modify elements inside `mode-line-position`, while it would be simpler, better, and easier, to just add its own element to `mode-line-position` (which is what the snippet above does). 

I'll try to think of a way to fix this in `sml`.

**Comment by wasamasa** [comment]
Thanks, I'll mail Tony Cubitt then to inform him about this if that's the way to go.

**Comment by Bruce-Connor** [comment]
Sure, let's see what he says..
He may be trying to do something specific with his approach. It looks to me like the above snippet achieves the same effect (other than the specific number), but I may just be missing something.

**Comment by wasamasa** [comment]
I did get a positive reply from the author, he wrote he'll put it in a future release since he agrees with the proposed change.

In other words, you can close this issue if you wish so or keep it open if you want to ponder whether such things can be "fixed" in a meaningful way.

**Comment by Bruce-Connor** [comment]
I have enough pondering to do in my life. :-)
Glad Toby agreed with it!

**Comment by tsc25** [comment]
For the record (in case another package author comes across this discussion, and wonders why `show-point-mode` originally manipulated the default mode-line directly, instead of using `mode-line-position`)...

The reason I didn't use `mode-line-position` is very banal: the `mode-line-position` variable didn't exist when I wrote show-point-mode! I've hardly touched the code since then (after all, `show-point-mode` is more of a useful one-liner than a package).

Thanks for suggesting this improvement. I've pushed this change to git and uploaded a new version of `show-point-mode` (0.3) to my web site.
